tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -320,15 +321,33 @@ class ApplicationInfo(
     if (this.appStart.nonEmpty) {
       val appStartNew: ArrayBuffer[ApplicationCase] = ArrayBuffer[ApplicationCase]()
       for (res <- this.appStart) {
-        val durationResult = ProfileUtils.OptionLongMinusLong(this.appEndTime, res.startTime)
+
+        val estimatedResult =
+          this.appEndTime match {
+            case Some(t) => this.appEndTime
+            case None =>
+              if (this.sqlEndTime.isEmpty && this.jobEndTime.isEmpty) {
+                None
```

Comment:
Just for my understanding. If both are empty then it throws NoSuchElementException?

Reply:
it returns None which is set to estimatedResult and that gets passed into OptionLongMinusLong which will just end up returning None for the durationResult which will result in null duration in table

Reply:
Thanks !